### PR TITLE
Tortoise conversion course selection

### PR DESCRIPTION
**Pull request by YuetingLiao**
**Issue**

Convert to tortoise for course-selection
Did not implement and test pytest for course-selection

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ node_modules/
 *.ipynb*
 courses20.xml
 .coverage
+src/api/tests/Dockerfile
```
Comment (lhain08) [inline .gitignore:13]: This should not be getting ignored here. If you did this to stop from pushing some kind of change to the dockerfile, you should prevent that by only adding files you wish to change.

File: src/api/db/student_course_selection.py [hunk 2]
```diff
@@ -1,45 +1,49 @@
-class student_course_selection:
-	def __init__(self, db_conn):
-		self.db_conn = db_conn
+from db.model import *
+from models.student_course_selection import StudentCourseSelection
```
Comment (lhain08) [inline src/api/db/student_course_selection.py:2]:
```suggestion
```

This import is not necessary

File: src/api/db/student_course_selection.py [hunk 3]
```diff
@@ -1,45 +1,49 @@
...
 					DELETE FROM
 						student_course_selection
 					WHERE
-						user_id = %s AND
-						semester = %s AND
-						course_name = %s AND
-						crn = %s
+						user_id = '%s' AND
+						semester = '%s' AND
+						course_name = '%s' AND
```
Comment (lhain08) [inline src/api/db/student_course_selection.py:38]: I could be wrong but I believe its possible for course names to have single quotes within them. We should probably escape this with the $<tag>$...$<tag>$ syntax as a precaution. You can see examples of this in the new db/courses.py file.

File: src/api/models/student_course_selection.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from tortoise import fields
+from tortoise.models import Model
+
+class StudentCourseSelection(Model):
+
+    user_id = fields.ForeignKeyField('user_account.user_id')
+    #user_id = Column(INTEGER, ForeignKey('user_account.user_id'))
```
Comment (lhain08) [inline src/api/models/student_course_selection.py:7]: Please remove commented code

File: src/api/app.py [hunk 5]
```diff
@@ -260,16 +260,29 @@ async def log_out(request: Request, session: SessionDeletePydantic):
...
-async def remove_student_course(request: Request, courseDelete:CourseDeletePydantic):
+@app.post('/api/user/course')
+async def add_student_course(request: Request, credentials: UserCoursePydantic):
     if 'user' not in request.session:
         return Response("Not authorized", status_code=403)
-    resp,error = course_select.remove_selection(courseDelete.name, courseDelete.semester, request.session['user']['user_id'], courseDelete.cid)
+    resp, error = await course_select.add_selection(credentials.name, credentials.semester, credentials.cid)
     return Response(status_code=200) if not error else Response(error, status_code=500)
+#
+#
```
Comment (lhain08) [inline src/api/app.py:270]: You need to finish/uncomment the delete and get methods. It also appears that there was some conflicts here as you have outdated versions of the methods as they are still using flask syntax in the commented versions. So make sure you get the correct version from TortoiseConversion when you do uncomment.

**Review by lhain08 — CHANGES_REQUESTED** [review]
I left a few comments, please let me know when you are done with them and I will take another look.

File: .gitignore (hunk 1 above)
Comment (YuetingLiao) [reply .gitignore:13]: Change has been discard

Comment (lhain08) [reply .gitignore:13]: This is still in the files changed

File: src/api/db/student_course_selection.py [hunk 6]
```diff
@@ -1,45 +1,48 @@
...
 					WHERE
-						user_id = %s AND
-						semester = %s AND
-						course_name = %s
+						user_id = '%s' AND
+						semester = '%s' AND
+						course_name = $course$%s$course$
 					"""
-			resp, error = self.db_conn.execute(sql, [uid, sem, name], False)
+			resp, error = self.db.execute(sql, [uid, sem, name], False)
```
Comment (lhain08) [inline src/api/db/student_course_selection.py:29]: Missing await here

**Review by lhain08 — CHANGES_REQUESTED** [review]

**Review by lhain08 — APPROVED** [review]

**Pull request merged**